src/libasr/pass/instantiate_template.cpp

```diff
@@ -26,16 +26,16 @@ class FunctionInstantiator : public ASR::BaseExprStmtDuplicator<FunctionInstanti
         new_func_name{new_func_name}
         {}
 
-    ASR::asr_t* instantiate_Function(ASR::Function_t &x) {
+    ASR::asr_t* instantiate_Function(ASR::Function_t *x) {
```

Comment:
What is the motivation of this change? Can `x` be `nullptr`?

Reply:
I think I made some changes that requires `&x` but then I made more changes that actually work with `*x`. I don't really have any reason for it. Should I revert it back?